pkg/social/common.go

```diff
@@ -23,24 +23,26 @@ func isEmailAllowed(email string, allowedDomains []string) bool {
 	return valid
 }
 
-func HttpGet(client *http.Client, url string) ([]byte, error) {
+func HttpGet(client *http.Client, url string) ([]byte, http.Header, error) {
```

Comment:
think returning an new struct (HttpGetResponse) with the body & headers is clearer